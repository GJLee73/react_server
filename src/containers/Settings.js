/**
 * 개인정보 변경 창.
 * @author G1
 * @logs // 18.2.27
 */

import React, { Component } from 'react';
import {connect} from 'react-redux'
import {SettingsComponent} from '../components'
import {getStatusRequest,getAllInfo} from '../modules/authentication'

export class Settings extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
        this.push=this.push.bind(this)
        this.quit=this.quit.bind(this)
    }

componentWillMount(){
    this.props.getAllInfo().then(
        ()=>{
        
        console.log('hi')
        }
    ).catch(()=>{
        console.log('g1')
        this.props.history.push('/')
    }
    )
    
 }
 /**
  * 타입에 따라 변경 페이지로 보내준다.
  * @param {string} param - 어떤 타입을 변경하는지 (nickname,password,phone,email) 
  */
push(param){
    if(param=='nickname')
    this.props.history.push('/settings/'+param)
    else{
        this.props.history.push('/pwdcheck/'+param)
    }
}
quit(){
   
    var confirmed = window.confirm('탈퇴하시겠습니까?')
    if(confirmed){
        this.props.history.push('/pwdcheck/quit')
    }
}
    render() { 
        return ( 
        <div>
         {this.props.status.get('isLoggedIn')? <SettingsComponent info={this.props.allInfo}
                                                                  push={this.push}
                                                                  quit={this.quit}/>:undefined }  
          
        </div> )
    }
}


const mapStateToProps = (state) =>{
	return{
        status: state.authentication.get('status'),
        allInfo: state.authentication.get('allInfo')
	};
};

const mapDispatchtoProps = (dispatch) => {
	return {
		getStatusRequest: ()=> {
			return  dispatch(getStatusRequest());
		},
		getAllInfo:()=>{
            return dispatch(getAllInfo())
        }
       
	};
}

export default connect(mapStateToProps, mapDispatchtoProps)(Settings);