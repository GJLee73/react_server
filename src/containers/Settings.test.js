import React from 'react';
import {shallow, mount} from 'enzyme'
import {Settings} from './Settings'
import {Map} from 'immutable'
import {MemoryRouter} from 'react-router-dom'

describe('Settings test',()=>{
    var getAllInfo = jest.fn(()=>Promise.reject())
    var history = {push:jest.fn()}
    global.confirm = jest.fn()
    var wrapper = shallow(<Settings getAllInfo={getAllInfo} history={history} status={Map({isLoggedIn:true})}/>)
    describe('componentWillMount test',()=>{
        it('getinfo fail',()=>{
            expect(getAllInfo).toHaveBeenCalledTimes(1)
            expect(history.push).toHaveBeenCalledTimes(1)
        })
    })
    describe('push test',()=>{
        it('param===nickname',()=>{
            wrapper.instance().push('nickname')
            expect(history.push.mock.calls[1][0]).toBe("/settings/nickname")
        })
        it('param==others',()=>{
            wrapper.instance().push('email')
            expect(history.push.mock.calls[2][0]).toBe('/pwdcheck/email')
        })
    })
    describe('quit test',()=>{
        wrapper.instance().quit()
        expect(global.confirm).toHaveBeenCalledTimes(1)
    }) 
    describe('snapShot test',()=>{
        it('logined snapshot',()=>{
            var mounted = mount(<Settings getAllInfo={getAllInfo} history={history} allInfo={Map({})} status={Map({isLoggedIn:true})}/>)
            expect(mounted).toMatchSnapshot()
        })
        it('not login snapshot',()=>{
            var mounted = mount(<Settings getAllInfo={getAllInfo} history={history}  status={Map({isLoggedIn:false})}/>)
            expect(mounted).toMatchSnapshot()
        })
    })
})