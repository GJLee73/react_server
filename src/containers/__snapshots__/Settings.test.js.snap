// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings test snapShot test logined snapshot 1`] = `
<Settings
  allInfo={
    Immutable.Map {
    }
  }
  getAllInfo={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  status={
    Immutable.Map {
      isLoggedIn: true,
    }
  }
>
  <div>
    <SettingsComponent
      info={
        Immutable.Map {
        }
      }
      push={[Function]}
      quit={[Function]}
    >
      <div>
        <div
          className="container auth"
        >
          <div
            className="card"
          >
            <div
              className="header blue white-text center"
            >
              <div
                className="card-content"
              >
                개인정보
              </div>
            </div>
            <div
              className="card-content"
            >
              <div
                className="row"
              >
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <div
                    className="input-field col s6"
                  >
                    닉네임:
                  </div>
                  <div
                    className="input-field col s6"
                  >
                    <div />
                  </div>
                </div>
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <div
                    className="input-field col s12"
                  >
                    비밀번호 수정
                  </div>
                </div>
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <div
                    className="input-field col s6"
                  >
                    이메일:
                  </div>
                  <div
                    className="input-field col s6"
                  >
                    <div />
                  </div>
                </div>
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <div
                    className="input-field col s6"
                  >
                    휴대폰:
                  </div>
                  <div
                    className="input-field col s6"
                  >
                    <div />
                  </div>
                </div>
              </div>
            </div>
            <button
              onClick={[Function]}
            >
              탈퇴하기
            </button>
          </div>
        </div>
      </div>
    </SettingsComponent>
  </div>
</Settings>
`;

exports[`Settings test snapShot test not login snapshot 1`] = `
<Settings
  getAllInfo={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  status={
    Immutable.Map {
      isLoggedIn: false,
    }
  }
>
  <div />
</Settings>
`;
